src/main/java/fi/metatavu/metaform/server/metaforms/ReplyController.java

```diff
@@ -287,6 +314,81 @@ public void deleteReply(Reply reply) {
     return anyReplyFieldDAO.listByReply(reply);
   }
 
+
+  /**
+   * Sets reply field value
+   * 
+   * @param reply reply
+   * @param name name 
+   * @param values values
+   * @return updated field
+   */
+  private ReplyField setReplyField(Reply reply, String name, List<?> values) {
+    ReplyField replyField = anyReplyFieldDAO.findByReplyAndName(reply, name);
+    if (replyField != null && !(replyField instanceof ListReplyField)) {
+      deleteField(replyField);
+      replyField = null;
+    }
+    
+    ListReplyField listReplyField = (ListReplyField) replyField;
+    if (values == null || values.isEmpty()) {
+      if (listReplyField != null) {
+        deleteListReplyFieldItems(listReplyField); 
+        listReplyFieldDAO.delete(listReplyField);
+      }
+      
+      return null;
+    }
+    
+    if (listReplyField == null) {
+      listReplyField = listReplyFieldDAO.create(UUID.randomUUID(), reply, name);
+    } else {
+      deleteListReplyFieldItems(listReplyField); 
+    }
+    
+    for (Object value : values) {
+      if (value != null) {
+        listReplyFieldItemDAO.create(UUID.randomUUID(), listReplyField, value.toString());
+      }
+    }
+    
+    return listReplyField;
+  }
+  
+  private void removeUnusedAttachmentReplyItems(List<AttachmentReplyFieldItem> items, List<String> savedFileRefs) {
+    items.stream()
+      .filter((attachmentReplyFieldItem) -> {
+        String fileRef = attachmentReplyFieldItem.getAttachment().getId().toString();
+        return !savedFileRefs.contains(fileRef);
+      })
+      .forEach(this::deleteAttachmentReplyFieldItem);
+  }
+
+  private void deleteAttachmentReplyFieldItem(AttachmentReplyFieldItem item) {
+    Attachment attachment = item.getAttachment();
+    attachmentReplyFieldItemDAO.delete(item);
+    attachmentController.deleteAttachment(attachment);
+  }
+  
+  private void attachUsedReplyItems(UUID userId, AttachmentReplyField field, List<AttachmentReplyFieldItem> items, List<String> savedFileRefs) {
+    List<String> usedFileRefs = items.stream()
+      .map(AttachmentReplyFieldItem::getAttachment)
+      .map(Attachment::getId)
+      .map(UUID::toString)
+      .collect(Collectors.toList());
+    
+    savedFileRefs.stream()
+      .filter((savedFileRef) -> !usedFileRefs.contains(savedFileRef))
+      .forEach((id) -> {
+        Attachment attachment = retrieveOrPersistAttachment(UUID.fromString(id), userId);
+        if (attachment != null) {
+          attachmentReplyFieldItemDAO.create(UUID.randomUUID(), field, attachment);
+        } else {
+          logger.error(String.format("Could not find attachment with id", id));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: String contains no format specifiers. ([squid:S3457](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS3457))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.metaform-api%3Ametaform-api&pullRequest=47&issues=AWVk3wGE4Bfu85TPTV1J&open=AWVk3wGE4Bfu85TPTV1J)

[ISSUE_KEY]: AWVk3wGE4Bfu85TPTV1J